src/GraphQL/Types/Scalars/FloatGraphType.cs

```diff
@@ -9,8 +9,10 @@ public class FloatGraphType : ScalarGraphType
         public override object ParseLiteral(IValue value) => value switch
         {
             FloatValue floatVal => floatVal.Value,
-            IntValue intVal => intVal.Value,
-            LongValue longVal => longVal.Value,
+            IntValue intVal => ParseValue(intVal.Value),
```

Comment:
```suggestion
            IntValue intVal => (double)intVal.Value,
```
I suggest the above and similar for performance to eliminate a dictionary lookup.  Note that for a widening conversion like this one, `checked` is not necessary, but nor should it create a performance impact as the compiler would know that.  Alternate syntax:
```suggestion
            IntValue intVal => checked((double)intVal.Value),
```